palantir-java-format/src/main/java/com/palantir/javaformat/java/java14/Java14InputAstVisitor.java

```diff
@@ -226,12 +231,30 @@ public Void visitCase(CaseTree node, Void unused) {
                 token("-");
                 token(">");
                 builder.space();
-                scan(node.getBody(), null);
+                if (node.getBody().getKind() == BLOCK) {
+                    visitBlock(
+                            (BlockTree) node.getBody(),
+                            CollapseEmptyOrNot.YES,
+                            AllowLeadingBlankLine.NO,
+                            AllowTrailingBlankLine.NO);
+                } else {
+                    scan(node.getBody(), null);
+                }
                 builder.guessToken(";");
                 break;
             default:
                 throw new AssertionError(node.getCaseKind());
         }
         return null;
     }
+
+    @Override
+    public Void visitLambdaExpression(LambdaExpressionTree node, Void unused) {
+        sync(node);
+        // Also format switch expressions as statement body instead of inlining them
+        boolean statementBody = node.getBodyKind() == LambdaExpressionTree.BodyKind.STATEMENT
+                || node.getBody().getKind() == Kind.SWITCH_EXPRESSION;
```

Comment:
Fix for case 1. This has to be in the `Java14InputAstVisitor.java` class to not break java 11 compatibility.

Reply:
should we add some kind of todo to collapse once we drop 11 (probably several years out)

Reply:
Added the todo.

But tbh, once we drop java 11 compat, we would collapse the whole `Java14InputAstVisitor` with the base class.